### PR TITLE
feat: keep selection state on scroll

### LINKED ISSUE #1020: Unable to keep block level selection state on scroll
Could be fixed by #1003

### PR DESCRIPTION
**Pull request by fundon**
Fixes: #1020, #1052

* [x] scroll smooth: a very simple implementation
* [x] resize
* [x] wheel: +/- selection rect
* [x] test cases
* [x] rect position consistency: #1085 
* [x] disable `SPACE`
* [x] fix `embed` rect position

### CODE REVIEW

File: pnpm-lock.yaml [hunk 1]
```diff
@@ -3942,14 +3940,13 @@ packages:
...
     dependencies:
-      '@typescript-eslint/parser': 5.49.0_et5x32uxl7z5ldub3ye5rhlyqm
       array-includes: 3.1.6
       array.prototype.flat: 1.3.0
       debug: 2.6.9
       doctrine: 2.1.0
       eslint: 8.32.0
       eslint-import-resolver-node: 0.3.6
-      eslint-module-utils: 2.7.4_qav77pk3harokpxo7diovtdrny
+      eslint-module-utils: 2.7.4_7gr2yud7moko5ggdmscvhh4rd4
```
Comment (himself65) [inline pnpm-lock.yaml]: these shouldn't be changed

File: packages/blocks/src/page-block/default/selection-manager.ts [hunk 2]
```diff
@@ -449,7 +449,7 @@ export class DefaultSelectionManager {
       // TODO: for the behavior of scrolling, see the native selection
       // speed easeOutQuad + easeInQuad
       if (Math.ceil(scrollTop) < max && clientHeight - y < this._thresold) {
-        // down
```
Comment (himself65) [inline packages/blocks/src/page-block/default/selection-manager.ts]: out of context, How to type "↓" on macOS?

File: packages/blocks/src/page-block/default/default-page-block.ts [hunk 3]
```diff
@@ -428,8 +463,9 @@ export class DefaultPageBlockComponent
       tryUpdateFrameSize(this.page, 1);
     });
 
-    // TMP: clear selected rects on scroll
-    document.addEventListener('wheel', this._clearSelection);
+    this.defaultViewportElement.addEventListener('wheel', this._onWheel);
+    this.defaultViewportElement.addEventListener('scroll', this._onScroll);
+    window.addEventListener('resize', this._onResize);
```
Comment (Innei) [inline packages/blocks/src/page-block/default/default-page-block.ts:468]: Maybe we can limit the current trigger frequency? Throttle or debounce

File: packages/blocks/src/page-block/default/default-page-block.ts [hunk 4]
```diff
@@ -148,8 +157,10 @@ export class DefaultPageBlockComponent
    */
   components: {
     dragHandle: DragHandle | null;
+    resizeObserver: ResizeObserver | null;
```
Comment (himself65) [inline packages/blocks/src/page-block/default/default-page-block.ts:160]: this shouldn't be here actually

File: tests/utils/actions/drag.ts [hunk 5]
```diff
@@ -63,10 +63,10 @@ export async function dragOverTitle(page: Page) {
   });
 }
 
-export async function dragEmbedResizeByBottomRight(page: Page) {
+export async function dragEmbedResizeByTopRight(page: Page) {
```
Comment (himself65) [inline tests/utils/actions/drag.ts:69]: Bottom overflow

File: packages/blocks/src/page-block/default/default-page-block.ts (hunk 4 above)
Comment (fundon) [reply packages/blocks/src/page-block/default/default-page-block.ts:160]: Do you have any suggestions?

Comment (himself65) [reply packages/blocks/src/page-block/default/default-page-block.ts:160]: let me do it in next pr

**Review by himself65 — APPROVED** [review]

**Comment by himself65** [comment]
codecov failed, ignore it

**Pull request merged**